google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Field.java

```diff
@@ -245,16 +98,31 @@ public Builder setName(String name) {
 
 
     /**
-     * Sets the value of the field.
+     * Sets the type of the field.
      *
      * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
      *     Data Types</a>
      */
-    public Builder setType(Type type) {
-      this.type = checkNotNull(type);
-      return this;
+    public Builder setType(LegacySQLTypeName type, Field... subFields) {
```

Comment:
It seems weird to require a "Legacy" SQL type name for non-legacy use cases.

Reply:
100% agree. Though it is an older problem, just made more visible with this PR. Before, there was a wrapper type, which "encapsulated" legacy type (basically one more wrapper around it, but it did nto have word "Legacy" in it). But the wrapper type was still using the legacy terminology (for example RECORD (legacy) instead of STRUCT (std)).

So I just removed a wrapper which was essentially doing nothing, it was not hiding the legacy nature of the types, since the legacy terminology was used. 

Currently we have LegacySQLTypeName referencing std types, so when you know legacy type, you can easy find a mapped std type. But it doesn't work in the opposite direction.

As we discussed offline switching to StandardSQLTypeName can be done in separate PR